crates/shared/src/zeroex_api.rs

```diff
@@ -75,6 +79,168 @@ impl SwapQuery {
     }
 }
 
+/// 0x API orders query parameters.
+///
+/// These parameters are currently incomplete, and missing parameters can be
+/// added incrementally as needed.
+/// https://0x.org/docs/api#signed-order
+#[derive(Clone, Debug)]
+pub struct OrdersQuery {
+    /// The address of the party that is allowed to fill the order.
+    /// If set to a specific party, the order cannot be filled by anyone else.
+    /// If left unspecified, anyone can fill the order.
+    pub taker: Option<H160>,
+    /// Allows the maker to enforce that the order flow through some
+    /// additional logic before it can be filled (e.g., a KYC whitelist).
+    pub sender: Option<H160>,
+    /// Address of the contract where the transaction should be sent,
+    /// usually this is the 0x exchange proxy contract.
+    pub verifying_contract: Option<H160>,
+}
+
+impl OrdersQuery {
+    /// Encodes the orders query as a url with parameters.
+    fn format_url(&self, base_url: &Url) -> Url {
+        let mut url = base_url
+            .join("/orderbook/v1/orders")
+            .expect("unexpectedly invalid URL segment");
+
+        if let Some(taker) = self.taker {
+            url.query_pairs_mut().append_pair("taker", &addr2str(taker));
+        }
+        if let Some(sender) = self.sender {
+            url.query_pairs_mut()
+                .append_pair("sender", &addr2str(sender));
+        }
+        if let Some(verifying_contract) = self.verifying_contract {
+            url.query_pairs_mut()
+                .append_pair("verifyingContract", &addr2str(verifying_contract));
+        }
+
+        url
+    }
+}
+
+impl Default for OrdersQuery {
+    fn default() -> Self {
+        Self {
+            taker: Some(H160::zero()),
+            sender: Some(H160::zero()),
+            verifying_contract: Some(DefaultZeroExApi::DEFAULT_VERIFICATION_CONTRACT),
+        }
+    }
+}
+
+#[derive(Clone, Derivative, Deserialize, PartialEq)]
+#[derivative(Debug)]
```

Comment:
Is this needed instead of derive Debug? Here and in following structs.

Reply:
I'm not sure. This is the result of copy-and-paste. 🙈  
I will check if it can be dropped.